src/module/actor/base.ts

```diff
@@ -359,10 +359,14 @@ class ActorPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e | n
 
     /** Get roll options from this actor's effects, traits, and other properties */
     getSelfRollOptions(prefix: "self" | "target" | "origin" = "self"): string[] {
-        const { rollOptions } = this;
-        return Object.keys(rollOptions.all).flatMap((o) =>
-            o.startsWith("self:") && rollOptions.all[o] ? o.replace(/^self/, prefix) : []
-        );
+        const { rollOptions, itemTypes } = this;
+        return Object.keys(rollOptions.all)
+            .flatMap((o) => (o.startsWith("self:") && rollOptions.all[o] ? o.replace(/^self/, prefix) : []))
+            .concat(
+                ...itemTypes.affliction
```

Comment:
Instead of adding item stuff to the actor roll options here, do it in the item itself. For example, check out AbstractEffectPF2e#prepareActorData()